### PR TITLE
Add CONTRIBUTING.md (renames BUILD.md)

### LINKED ISSUE #1692: Change the name of BUILD.md to CONTRIBUTING.md
## 1. General summary of the issue

Suggest changing the name of BUILD.md to CONTRIBUTING.md to be more discoverable for contributors. I was looking for a CONTRIBUTING.md when trying to figure out how to build the project, I figured it out before I found the BUILD.md file. *Also GitHub automatically provide links to the CONTRIBUTING.md in certain places if it exist.*

We should also add that to be able to build the .dll's there is a need to run `.\build.ps1 -Clean`.

Happy to contribute this if it is acceptable.

### PR DESCRIPTION
**Pull request by johlju**
## 1. General summary of the pull request

- Renames BUILD.md to CONTRIBUTING.md to be more discoverable for contributors.
- Updated text around how to build the module.

Fixes #1692

---
This change is [Reviewable](https://reviewable.io/reviews/pester/pester/1697)

### CODE REVIEW

**Comment by fflaten** [comment]
Maybe add a note below the import module line that changes to c# will require a new powershell session (restart terminal) if Pester was previously imported. Assemblies can't be unloaded/reloaded even with `-Force`.

**Comment by johlju** [comment]
> Maybe add a note below the import module line that changes to c# will require a new powershell session (restart terminal) if Pester was previously imported. Assemblies can't be unloaded/reloaded even with `-Force`.

Yes, good suggestion, will push a change when I get home.

**Comment by nohwnd** [comment]
@johlju please wait till this one is merged, or propose change on that PR to and the note about re-loading pester after clean build. https://github.com/pester/Pester/pull/1647

**Comment by johlju** [comment]
@nohwnd I can wait until that PR is through and then I will rebase this one. I start tracking the other PR so I see when it gets merged.

**Comment by nohwnd** [comment]
@johlju the other PR was just merged.

**Comment by johlju** [comment]
@nohwnd the PR is rebased. I updated the text somewhat. Let me know if you want me to change anything.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ To skip P tests:
 
 ```powershell
 .PARAMETER CI
-  Exits after run.  Enables test results and code coverage on `/src/*`. Enable exit with 1 if tests don't pass. Forces P Tests to fail when `dt` is left in the tests. `dt` only runs the specified test, so leaving it in code would run only one test from the file on the server.
+  Exits after run. Enables test results and code coverage on `/src/*`. Enable exit with 1 if tests don't pass. Forces P Tests to fail when `dt` is left in the tests. `dt` only runs the specified test, so leaving it in code would run only one test from the file on the server.
```
Comment (fflaten) [inline CONTRIBUTING.md:80]: > Enable exit with 1 if tests don't pass

This is just me nitpicking, but failed P-tests returns 1. Failed tests during the normal Pester tests that run after will result in a value greater than zero (sum of failed tests, blocks++). Maybe "_Enables exit with non-zero exit code if tests don't pass._" ?

**Comment by fflaten** [comment]
Should the comment-based help be moved to `build.ps1` and `test.ps1` as part of this PR? Already got most of it ready for `test.ps1` (remember PARAMETER prefix on all btw). Then the documentation could simply reference the script help.  `.\build.ps1 -?`, `Get-Help .\build.ps1 -Full` etc.

**Comment by johlju** [comment]
@fflaten I agree the section "### Test Parameters" could be moved to comment-based help in test.ps1.

**Comment by johlju** [comment]
@nohwnd please double check the comment-based help so it is correct. 🙂

**Comment by johlju** [comment]
@fflaten I saw now that you are the reviewer, please double-check the comment-based help. 🙂

**Comment by fflaten** [comment]
I personally like it 👍

Considering the parameter-help for tests.ps1 is moved outside the file, maybe we should mention what P tests are in a few words before telling how to skip them? E.g.

> test.ps1 runs two sets of tests. It first runs P tests written in the P module which are used for unit testing the runtime itself and acceptance tests of Pester. Then it runs normal Pester tests for all functions in the module. <something like.. when modifying a single function, assertion operator ++ you might want to skip the P tests to save time. This is done using:>

Or remove that completely and let them find it in the help?

**Comment by nohwnd** [comment]
@fflaten up to you. I am okay with it either way.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -17,7 +59,7 @@ Pester has a C# Solution which requires .NET Framework SDKs and Developer Packs
 
 ## Running Tests
 
-In Powershell, run test.ps1.  This defines the inherited function InPesterModuleScope and some types required for the tests.
+In Powershell, run test.ps1. This defines the inherited function InPesterModuleScope and some types required for the tests.
```
Comment (fflaten) [inline CONTRIBUTING.md:62]:
```suggestion
Pester uses two types of tests:
 
1. P tests (`*.ts.ps1`) written in the P module. These are used for unit testing the runtime itself and running acceptance tests for Pester
2. Pester tests (`*.tests.ps1`) for all functions in the module. 

In Powershell, use `test.ps1`. The scripts runs a build and imports required helper-functions like `InPesterModuleScope` before starting.
```

**Review by fflaten — COMMENTED** [review]
I think P tests are mystical for most new contributors, so it wouldn't hurt to give a quick intro to get people up to speed. How about this?

File: CONTRIBUTING.md (hunk 2 above)
Comment (nohwnd) [reply CONTRIBUTING.md]: LGTM

**Comment by nohwnd** [comment]
Is this ready to merge?

**Comment by johlju** [comment]
If there are no more comments it’s good to go for me.

**Pull request merged**